docs/fastdds/dds_layer/core/status/status.rst

```diff
@@ -430,33 +430,34 @@ List of status data members:
   Handle to the last instance whose sample was rejected.
   If no sample was ever rejected, it will have value ``c_InstanceHandle_Unknown``.
 
-.. warning::
-
-    Currently this status is not supported and will be implemented in future releases.
-    As a result, trying to access this status will return ``NOT_SUPPORTED``
-    and the corresponding listener will never be called.
-
-
 .. _dds_layer_core_status_sampleRejectedStatusKind:
 
 SampleRejectedStatusKind
 ^^^^^^^^^^^^^^^^^^^^^^^^
 
-There are four possible values (see |SampleRejectedStatusKind-api|):
-
-* |NOT_REJECTED|:
-  It means there have been no rejections so far on this DataReader.
-  The rejection reason will have this value only while the total count of rejections is zero.
-* |REJECTED_BY_INSTANCES_LIMIT|:
-  The sample was rejected because the
-  :ref:`max_instances<resourcelimitsqospolicy>` limit was reached.
-* |REJECTED_BY_SAMPLES_LIMIT|:
-  The sample was rejected because the
-  :ref:`max_samples<resourcelimitsqospolicy>` limit was reached.
-* |REJECTED_BY_SAMPLES_PER_INSTANCE_LIMIT|:
-  The sample was rejected because the
-  :ref:`max_samples_per_instance<resourcelimitsqospolicy>` limit was reached.
+In Fast DDS, samples can be rejected due to resource limit reasons.
+However, the fact that the samples are rejected does not imply that they are lost, i.e. a rejected sample may be
+accepted in the future.
+
+|SampleRejectedStatusKind-api| specifies the reason of the rejection:
+
+* |NOT_REJECTED| specifies the samples were not rejected.
+
+* |REJECTED_BY_SAMPLES_LIMIT| specifies the samples were rejected because there is no enough resources to stored them
+  and the system should guarantee the free resources are available for other samples.
+  This situation occurs at lower RTPS communication layer and it happens when there are still samples with lower
+  sequence number without being received and there is no enough resources for all of them because |max_samples-api| was
+  reached.
+
+* |REJECTED_BY_INSTANCES_LIMIT| specifies the samples were rejected because there is no enough resources to allocate
+  their instances.
+  This situation occurs at DDS layer, in the DataReader's history, when it should reserve resources for new
+  instances but the history's resources reach |max_instances-api|.
 
+* |REJECTED_BY_SAMPLES_PER_INSTANCE_LIMIT| specifies the samples were rejected because there is no enough resources in
+  their instance to stored them.
+  This situation occurs at DDS layer, in the DataReader's history, when it is configured with |KEEP_ALL_HISTORY_QOS-api|
+  and the instance's resources reach |max_samples_per_instance-api|.
```

Comment:
```suggestion
* |REJECTED_BY_SAMPLES_PER_INSTANCE_LIMIT| specifies that the samples were rejected because there were not enough resources within
  their instance to stored them.
  This situation, which arises in the DDS layer, more precisely in the :class:`DataReader`'s history, occurs when the :class:`DataReader` is configured with |KEEP_ALL_HISTORY_QOS-api|
  and the instance's number of samples has reached |max_samples_per_instance-api|.
```